src/components/structures/MessagePanel.tsx

```diff
@@ -1105,28 +1106,30 @@ abstract class BaseGrouper {
 // Grouping only events sent by the same user that sent the `m.room.create` and only until
 // the first non-state event, beacon_info event or membership event which is not regarding the sender of the `m.room.create` event
 class CreationGrouper extends BaseGrouper {
-    public static canStartGroup = function (panel: MessagePanel, ev: MatrixEvent): boolean {
+    public static canStartGroup = function (
+        panel: MessagePanel,
+        { event: ev, shouldShow }: EventAndShouldShow,
+    ): boolean {
         return ev.getType() === EventType.RoomCreate;
     };
 
-    public shouldGroup(ev: MatrixEvent): boolean {
+    public shouldGroup({ event: ev, shouldShow }: EventAndShouldShow): boolean {
         const panel = this.panel;
         const createEvent = this.event;
```

Comment:
given you don't use the shouldShow flag, I'd find this clearer:

```suggestion
        const createEvent = this.event.event;
```

Reply:
Done, ta